File: frontend/src/types/ingestion.ts
export interface IngestionJob {
  id: number;
  source_name: string;
  status: 'Pending' | 'Processing' | 'Completed' | 'Failed' | string;
  total_records: number;
  processed_records: number;
  failed_records: number;
  progress_percentage: number;
  error_log?: string;
  started_at: string;
  completed_at?: string;
}

export interface AdminStats {
  total_cars: number;
  total_models: number;
  total_manufacturers: number;
  total_sources: number;
  total_vector_docs: number;
  vector_store_type: string;
}

File: backend/app/models/ingestion.py
from datetime import datetime, timezone
from typing import Optional
from sqlalchemy import String, DateTime, Float, Integer, Text
from sqlalchemy.orm import Mapped, mapped_column
from app.db.session import Base

class IngestionJob(Base):
    __tablename__ = "ingestion_jobs"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, index=True)
    source_name: Mapped[str] = mapped_column(String(255), nullable=False)
    status: Mapped[str] = mapped_column(String(50), default="Pending") # Pending, Processing, Completed, Failed
    total_records: Mapped[int] = mapped_column(Integer, default=0)
    processed_records: Mapped[int] = mapped_column(Integer, default=0)
    failed_records: Mapped[int] = mapped_column(Integer, default=0)
    progress_percentage: Mapped[float] = mapped_column(Float, default=0.0)
    error_log: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    started_at: Mapped[datetime] = mapped_column(DateTime, default=lambda: datetime.now(timezone.utc))
    completed_at: Mapped[Optional[datetime]] = mapped_column(DateTime, nullable=True)
